### ISSUE TITLE
Issue with Executing SQL Commands with Semicolons in Init-SQL-Commands for Flight SQL Server

### ISSUE THREAD

**Issue by rohitprg** [voltrondata/flight-sql-server-example#33]
The option --init-sql-commands of flight_sql_server uses semicolon (';') as a delimiter to segregate the SQL command, which is then run on the server. If a SQL command, such as a connection string containing a semicolon, is used, it would be divided into distinct commands at each semicolon. This results in an incorrect operation, as a single command is erroneously divided into multiple commands.

```sql
CREATE SECRET (
    TYPE AZURE,
    CONNECTION_STRING 'DefaultEndpointsProtocol=https;AccountName=***;AccountKey=*******==;EndpointSuffix=***'
)
```  
Proposed solution
In order to escape the semicolons in the connection string, they could be substituted with a unique character sequence (like `"_semicolon_"`). Subsequently, in the `RUN_INIT_COMMANDS` macro, this sequence can be reverted to a semicolon prior to the execution of the command.

Here is how macro could be altered:

```cpp
#define RUN_INIT_COMMANDS(serverType, init_sql_commands) \
    do { \
        if (init_sql_commands != "") { \
            std::vector<std::string> tokens; \
            boost::split(tokens, init_sql_commands, boost::is_any_of(";")); \
            for (std::string &init_sql_command: tokens) { \
                if (init_sql_command.empty()) continue; \
                boost::replace_all(init_sql_command, "_semicolon_", ";"); \
                std::cout << "Running Init SQL command: " << std::endl << init_sql_command << ";" << std::endl; \
                ARROW_RETURN_NOT_OK(serverType->ExecuteSql(init_sql_command)); \
            } \
        } \
    } while (false)
```

And this is how the semicolons in the SQL command can be escaped:

```sql
CREATE SECRET (
    TYPE AZURE,
    CONNECTION_STRING 'DefaultEndpointsProtocol=https_semicolon_AccountName=***_semicolon_AccountKey=*******==_semicolon_EndpointSuffix=***'
)
```

**Comment by prmoore77** [comment]
@rohitprg - what are your thoughts about us allowing escaping the semi-colon with a `\` character?  Thanks for your help.

**Comment by rohitprg** [comment]
> what are your thoughts about us allowing escaping the semi-colon with a `\` character?

Allowing the escaping of the semicolon with a backslash (\) character is a good idea, particularly because it is a common way to handle such situations.

**Comment by prmoore77** [comment]
hey @rohitprg - could you kindly see if your issue was truly resolved by the latest release: `v1.3.1` ?  Thanks.

**Comment by prmoore77** [comment]
> hey @rohitprg - could you kindly see if your issue was truly resolved by the latest release: `v1.3.1` ? Thanks.

Please note - the fix was to NOT delimit on semi-colons which are contained within single quotes.  There is no need to escape the semi-colon(s)...

**Comment by rohitprg** [comment]
@prmoore77 -  I have verified the fix and it's functioning properly now. Thanks for the prompt fix. 
It appears that in the modified code, semicolons within single quotes are not treated as command separators.
